tests/contrib/test_dictionary_storage.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for oauth2client.dictionary_storage"""
+
+import unittest2
+
+from oauth2client import GOOGLE_TOKEN_URI
+from oauth2client.client import OAuth2Credentials
+from oauth2client.contrib.dictionary_storage import DictionaryStorage
+
+
+class DictionaryStorageTests(unittest2.TestCase):
+
+    def _generate_credentials(self, scopes=None):
+        return OAuth2Credentials(
+            'access_tokenz',
+            'client_idz',
+            'client_secretz',
+            'refresh_tokenz',
+            '3600',
+            GOOGLE_TOKEN_URI,
+            'Test',
+            id_token={
+                'sub': '123',
+                'email': 'user@example.com'
+            },
+            scopes=scopes)
+
+    def test_constructor(self):
+        dictionary = {}
+        key = 'test-key'
+        storage = DictionaryStorage(dictionary, key)
+
+        self.assertEqual(dictionary, storage._dictionary)
+        self.assertEqual(key, storage._key)
+        self.assertIsNone(storage._lock)
+
+        lock = object()
+        storage = DictionaryStorage(dictionary, key, lock=lock)
+        self.assertEqual(storage._lock, lock)
+
+    def test_get(self):
+        credentials = self._generate_credentials()
+        dictionary = {}
+        key = 'credentials'
+        storage = DictionaryStorage(dictionary, key)
+
+        self.assertIsNone(storage.get())
+
+        dictionary[key] = credentials.to_json()
+
+        self.assertEqual(storage.get().access_token, credentials.access_token)
```

Comment:
Made it more comprehensive by testing the access, id, and refresh tokens (the "important" bits) and the client id. Testing any more and I'm basically just re-testing `Credentials.to_json()` and `from_json()` which presumably have their own tests.